miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultInstrumentService.java

```diff
@@ -71,9 +77,74 @@ public long create(Instrument instrument) throws IOException {
     return save(instrument);
   }
 
+  @Override
+  public void addServiceRecord(ServiceRecord record, Instrument instrument) throws IOException {
+    // originally located at HibernateServiceRecordDao
+    if (instrument.getDateCommissioned() != null) {
+      throw new IOException("Cannot add service records to a retired instrument!");
```

Comment:
Since this has moved to the service, it would be more appropriate to throw a ValidationException here - this will give the user an informative error on the page they were working on, instead of a "Sorry, something unexpected has occurred" 500-series error.